src/c++/library/http_client.h

```diff
@@ -99,10 +143,13 @@ class InferenceServerHttpClient : public InferenceServerClient {
   /// base path in the following format: host:port/<base-path>.
   /// \param verbose If true generate verbose output when contacting
   /// the inference server.
+  /// \param ssl_options Specifies the settings for configuring
+  /// SSL encryption and authorization.
   /// \return Error object indicating success or failure.
   static Error Create(
       std::unique_ptr<InferenceServerHttpClient>* client,
-      const std::string& server_url, bool verbose = false);
+      const std::string& server_url, bool verbose = false,
+      const HttpSslOptions& ssl_options = HttpSslOptions());
```

Comment:
should this also have a `bool use_ssl = false` like the grpc client create function?

Reply:
The libcurl requires the `https://` (scheme) to be part of the url to detect the TLS/SSL connection. We can internally add the `https://` based on another parameter `use_ssl`, but I decided to let users explicitly mention the scheme in the url just to be extra clear.